PySDM/backends/impl_thrust_rtc/test_helpers/cpp2python.py

```diff
@@ -189,6 +189,7 @@ def to_numba(name, args, iter_var, body):
     result = (
         f"""
 def make(self):
+    import math
```

Comment:
- this will not work on GPU
- not covered with tests

Reply:
What to do then?

Reply:
Here's how the problem looks like (when not using `FakeThrust`, but when actually compiling for CUDA):

```
...
Errors:
header_of_structs.h(501): error: identifier "math" is undefined

1 error detected in the compilation of "saxpy.cu".
...
```

```
FAILED tests/unit_tests/dynamics/collisions/test_fragmentations.py::TestFragmentations::test_fragmentation_fn_call[ThrustRTC-fragmentation_fn3] - SystemError: An internal error happend
FAILED tests/unit_tests/dynamics/collisions/test_fragmentations.py::TestFragmentations::test_fragmentation_fn_call[ThrustRTC-fragmentation_fn5] - SystemError: An internal error happend
FAILED tests/unit_tests/dynamics/collisions/test_fragmentations.py::TestFragmentations::test_fragmentation_limiters_vmin[ThrustRTC-fragmentation_fn2] - SystemError: An internal error happend
FAILED tests/unit_tests/dynamics/collisions/test_fragmentations.py::TestFragmentations::test_fragmentation_limiters_vmin[ThrustRTC-fragmentation_fn4] - SystemError: An internal error happend
FAILED tests/unit_tests/dynamics/collisions/test_fragmentations.py::TestFragmentations::test_fragmentation_limiters_vmax[ThrustRTC-fragmentation_fn2] - SystemError: An internal error happend
FAILED tests/unit_tests/dynamics/collisions/test_fragmentations.py::TestFragmentations::test_fragmentation_limiters_vmax[ThrustRTC-fragmentation_fn4] - SystemError: An internal error happend
FAILED tests/unit_tests/dynamics/collisions/test_fragmentations.py::TestFragmentations::test_fragmentation_limiters_nfmax[ThrustRTC-fragmentation_fn2] - SystemError: An internal error happend
FAILED tests/unit_tests/dynamics/collisions/test_fragmentations.py::TestFragmentations::test_fragmentation_limiters_nfmax[ThrustRTC-fragmentation_fn4] - SystemError: An internal error happend
=============================================================================== 8 failed, 34 passed, 6 xfailed in 107.66s (0:01:47) ===============================================================================

```

Reply:
here's one fix: https://github.com/edejong-caltech/PySDM/commit/07c90b0d5cb286cc56a7403833b18d8a20991f69

Reply:
and several other GPU-compatibility issues also addressed in follow-up commits